nemo/collections/asr/models/eesd_models.py

```diff
@@ -643,7 +635,7 @@ def sort_targets_with_preds_new(self, labels, preds, noise=0):
         match_loss = torch.zeros(labels.shape[0], perm_size)
         for k in range(labels.shape[0]):
             for i in range(perm_size):
-                match_loss[k,i] = self.loss(probs=preds[k,:,:].unsqueeze(0), labels=permed_labels[k,:,i,:].unsqueeze(0), signal_lengths=torch.tensor([labels.shape[1]], dtype=torch.int32))
+                match_loss[k,i] = self.loss(probs=preds[k,:,:].unsqueeze(0), labels=permed_labels[k,:,i,:].unsqueeze(0), target_lens=torch.tensor([labels.shape[1]], dtype=torch.int32))
```

Comment:
The `target_lens` should be the actual length of targets before zero padding when computing the bce loss. 
`label.shape[1]` is the length after zero padding. 
Maybe we also need check other functions where `label.shape[1]` is used.